streams/src/main/java/org/apache/kafka/streams/kstream/internals/metrics/Sensors.java

```diff
@@ -39,4 +44,37 @@ public static Sensor lateRecordDropSensor(final InternalProcessorContext context
         );
         return sensor;
     }
+
+    public static Sensor recordLatenessSensor(final InternalProcessorContext context) {
+        final StreamsMetricsImpl metrics = context.metrics();
+
+        final Sensor sensor = metrics.nodeLevelSensor(
+            context.taskId().toString(),
+            context.currentNode().name(),
+            "record-lateness",
+            Sensor.RecordingLevel.INFO
```

Comment:
Could this be a `DEBUG` level recording? Again no strong opinion here.

Reply:
I think the average lateness is actually one of the more important metrics for determining what timeout to use for suppression.

If you're worried about the performance, I can tackle KAFKA-7275 sooner rather than later.

Reply:
1) Current `lateRecordDropSensor` seems to be completely paired with `recordLatenessSensor`, i.e. there is no place where we only record one of them but not both. It makes me thinking if we can optimize this a bit to get one sensor, with a COUNT / and AVG / MAX metrics with it, as it seems that `lateRecordDropSensor` is just a COUNT of the `recordLatenessSensor`.

2) I'm also a bit concerned about the performance as it is a per task-processor-node-level sensor. Would better consider doing KAFKA-7275 sooner than later.

Reply:
1. I think you've actually pointed out a bug in my measurement. I intended to measure record lateness regardless of whether it was *too late* or not. I'll fix it.

2. Ok, I'll consider KAFKA-7275 asap.